server/app/controllers/CiviFormController.java

```diff
@@ -39,4 +40,19 @@ protected CompletableFuture<Void> checkProgramAdminAuthorization(
         .orElseThrow()
         .checkProgramAuthorization(programName, request);
   }
+
+  /** Checks that the profile is authorized to access the specified program. */
+  protected CompletableFuture<Void> checkProgramAuthorization(
+      ProfileUtils profileUtils,
+      VersionRepository versionRepository,
```

Comment:
**suggestion (blocking):** Make `profileUtils` and `versionRepository` members of `CivFormController` and pass them via `super(profileUtils, versionRepository)` in the child class c'tors instead of passing them to each method call.

Reply:
Done